src/components/light.js

```diff
@@ -71,6 +105,15 @@ module.exports.Component = registerComponent('light', {
       this.light.el = el;
       el.setObject3D('light', this.light);
     }
+
+    if (!data.target && data.type === 'spot') {
```

Comment:
I think this and below should be done within `getLight` which handles creation of the light?